### ISSUE TITLE
XY Chart: Create our own chart library

### ISSUE THREAD

**Issue by tahini** [theia-ide/theia-trace-extension#42]
XY Component is using ChartJs wrapped in a react-chartjs-2 library. We should create our own instead of using the library. It's not hard to do and will offer way more flexibility to change the axis, draw over the chart and add plugins to chartjs. This should be looked into before subsequent tasks as it may have an impact on how to solve them. Some problems occur when panning. Observe what is happening, that shouldn’t be.

**Comment by MatthewKhouzam** [comment]
We appear to be moving towards D3. @Rodrigoplp-work does this still make sense?

**Comment by Rodrigoplp-work** [comment]
Yes, absolutely! I totally agree. This goes right with what is discussed in the "0003-D3-charts" ADR.
With the examples that were included in the ADR, plus the work made for the y-axis recently, we are moving really close to being able to make this transition.
